plugins/inmemoryscanner/src/lib/InMemory.cpp

```diff
@@ -20,29 +21,28 @@ VMI_PLUGIN_API_VERSION_INFO
 
 using VmiCore::ActiveProcessInformation;
 using VmiCore::Plugin::IPluginConfig;
-using VmiCore::Plugin::LogLevel;
 using VmiCore::Plugin::PluginInterface;
 
 namespace InMemoryScanner
 {
     PluginInterface* pluginInterface;
     std::unique_ptr<Scanner> scanner;
+    std::unique_ptr<VmiCore::ILogger> staticLogger;
 
     void shutdownCallback()
     {
-        pluginInterface->logMessage(LogLevel::info, LOG_FILENAME, "Shutdown initiated");
+        staticLogger->info("Shutdown initiated");
         try
         {
             scanner->scanAllProcesses();
         }
         catch (const std::exception& exc)
         {
-            auto errorMsg = "Error occurred during shutdown scan. Reason: " + std::string(exc.what());
-            pluginInterface->logMessage(LogLevel::error, LOG_FILENAME, errorMsg);
-            pluginInterface->sendErrorEvent(errorMsg);
+            staticLogger->error("Error occurred during shutdown scan", {{"Exception", exc.what()}});
+            pluginInterface->sendErrorEvent(exc.what());
```

Comment:
You are sure that just the exception is sufficient here?

Reply:
This is how it has been done in vmicore. I've simply "streamlined" the behavior. I suppose it's fine. If you want to have more information you can always have a look at the logs.